### PR TITLE
docs(samples): added samples for Mute config

### PR DESCRIPTION
**Pull request by Sita04**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-securitycenter/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by Sita04** [comment]
@shubha-rajan PTAL

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      BulkMuteFindingsRequest bulkMuteFindingsRequest =
+          BulkMuteFindingsRequest.newBuilder()
+              .setParent(parentPath)
+              // To create mute rules, see:
+              // https://cloud.google.com/security-command-center/docs/how-to-mute-findings#create_mute_rules
+              .setFilter(muteRule)
+              .build();
+
+      BulkMuteFindingsResponse response =
+          client.bulkMuteFindingsAsync(bulkMuteFindingsRequest).get();
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java:58]: what if there is an error? how do we handle it? what exception do we throw? what error message can we provide?

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      CreateMuteConfigRequest request =
+          CreateMuteConfigRequest.newBuilder()
+              .setParent(parentPath)
+              // Set a random id; max of 63 chars.
+              .setMuteConfigId(muteConfigId)
+              .setMuteConfig(muteConfig)
+              .build();
+
+      MuteConfig response = client.createMuteConfig(request);
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java:66]: please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Deletes a mute configuration given its resource name.
+  // Note: Previously muted findings are not affected when a mute config is deleted.
+  public static void deleteMuteRule(String projectId, String muteConfigId) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+      // Use appropriate MuteConfigName methods depending upon the type of parent.
+      // (org -> MuteConfigName.ofOrganizationMuteConfigName()
+      // folder -> MuteConfigName.ofFolderMuteConfigName()
+      // project -> MuteConfigName.ofProjectMuteConfigName)
+      client.deleteMuteConfig(MuteConfigName.ofProjectMuteConfigName(projectId, muteConfigId));
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java:43]: same as above, please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/GetMuteRule.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Retrieves a mute configuration given its resource name.
+  public static void getMuteRule(String projectId, String muteConfigId) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+      // Use appropriate MuteConfigName methods depending upon the type of parent.
+      // (org -> MuteConfigName.ofOrganizationMuteConfigName()
+      // folder -> MuteConfigName.ofFolderMuteConfigName()
+      // project -> MuteConfigName.ofProjectMuteConfigName)
+      MuteConfig muteConfig =
+          client.getMuteConfig(MuteConfigName.ofProjectMuteConfigName(projectId, muteConfigId));
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/GetMuteRule.java:44]: please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  // Similarly, listing configs at folder level will list all the configs
+  // at the folder and project levels.
+  public static void listMuteRules(String parent) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+
+      ListMuteConfigsRequest listMuteConfigsRequest =
+          ListMuteConfigsRequest.newBuilder().setParent(parent).build();
+
+      // List all Mute Configs present in the resource.
+      for (MuteConfig muteConfig : client.listMuteConfigs(listMuteConfigsRequest).iterateAll()) {
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java:49]: please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/SetMuteUnmuteFinding.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  // If a finding is already muted, muting it again has no effect.
+  // Similarly, unmuting a finding that isn't muted has no effect.
+  // Various mute states are: MUTE_UNSPECIFIED/MUTE/UNMUTE.
+  public static void setMute(String findingPath) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+
+      SetMuteRequest setMuteRequest =
+          SetMuteRequest.newBuilder().setName(findingPath).setMute(Mute.MUTED).build();
+
+      Finding finding = client.setMute(setMuteRequest);
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/SetMuteUnmuteFinding.java:51]: please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+              // Set the update mask to specify which properties of the Mute Config should be
+              // updated.
+              // If empty, all mutable fields will be updated.
+              // Make sure that the mask fields match the properties changed in 'updateMuteConfig'.
+              // For more info on constructing update mask path, see the proto or:
+              // https://cloud.google.com/security-command-center/docs/reference/rest/v1/folders.muteConfigs/patch?hl=en#query-parameters
+              .setUpdateMask(FieldMask.newBuilder().addPaths("description").build())
+              .build();
+
+      MuteConfig response = securityCenterClient.updateMuteConfig(updateMuteConfigRequest);
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java:63]: please consider showing error handling.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      System.out.println(response);
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java:64]: does this automatically provide response.toString?

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+
+      ListFindingsRequest request = ListFindingsRequest.newBuilder().setParent(sourceName).build();
+
+      return client.listFindings(request);
+    }
+  }
+
+  private static String getOrganizationId() {
+    return "1081635000895";
```
Comment (stephaniewang526) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:172]: It's better to save this as an env var and store it in java-docs-samples-testing project's secret manager.

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+  }
+
+  private static String getProject() {
+    return "project-a-id";
```
Comment (stephaniewang526) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:176]: this should also be an env var which we pull from secret manager.

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void afterEach() {
+    stdOut = null;
+    System.setOut(null);
+  }
+
```
Comment (stephaniewang526) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:190]: Where are you cleaning up all the test resources you created for these samples? Usually we do it in `tearDown` like [this](https://github.com/googleapis/java-bigquery/blob/c37f800941f70d49e3316bd5f4299f3fce654f64/samples/snippets/src/test/java/com/example/bigquery/CreateDatasetIT.java#L63).

**Review by stephaniewang526 — CHANGES_REQUESTED** [review]

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java (hunk 1 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java:58]: The client doesn't provide Operation Metadata. Hence, error handling is intrinsic and exception will be thrown. Same is the case for other APIs as well.
I've added a comment for more context.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java (hunk 2 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java:66]: The client response has no error metadata and only exceptions are thrown.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java (hunk 3 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java:43]: The API call is non-LRO and returns an Empty response.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/GetMuteRule.java (hunk 4 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/GetMuteRule.java:44]: The API is not LRO and hence the client response has no error metadata and only exceptions are thrown.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java (hunk 5 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java:49]: The client response has no error metadata and only exceptions are thrown.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/SetMuteUnmuteFinding.java (hunk 6 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/SetMuteUnmuteFinding.java:51]: The client response has no error metadata and only exceptions are thrown.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java (hunk 7 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java:63]: This is a non-LRO operation and hence no metadata is provided by the client.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java (hunk 8 above)
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java:64]: Yes, it does :)

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java (hunk 11 above)
Comment (Sita04) [reply samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:190]: Please find them here: https://github.com/googleapis/java-securitycenter/pull/719/files#diff-f54e46cf1d9d8390ef2a0bc3feeaacebf6d5a20b63f238405a8f476b7f392523R97-R105
The only deletable resources for these samples are Mute rules.

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+      return response;
+    }
+  }
+
+  public static Finding createFinding(String sourceName, String findingId) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+
+      Instant eventTime = Instant.now();
+
+      // The resource this finding applies to.  The Cloud Security Command Center UI can link
```
Comment (catherinerosesantos) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:133]: Use only 1 space after a period.

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+      // the findings for a resource to the corresponding Asset of a resource
+      // if there are matches.
+      // TODO(Developer): Replace the below sample resource name
+      String resourceName = "//cloudresourcemanager.googleapis.com/organizations/11232";
+
+      // Start setting up a request to create a finding in a source.
```
Comment (catherinerosesantos) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:139]: Start setting up --> Set up

**Review by catherinerosesantos — CHANGES_REQUESTED** [review]
Looks great! Just a few minor suggestions.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+    // - organizations/{organization}/muteConfigs/{config_id}
+    // - folders/{folder}/muteConfigs/{config_id}
+    // - projects/{project}/muteConfigs/{config_id}
+    // TODO: Replace the variables within {}
+    String muteConfigName = "{any-one-of-the-above-formats}";
+    updateMuteRule(muteConfigName);
+  }
+
+  // Updates an existing mute configuration.
+  // The following can be updated in a mute config: description, and filter/ mute rule.
```
[the remark this answers is not in the document]
Comment (Sita04) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/UpdateMuteRule.java:41]: Agreed. Thanks.

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java (hunk 1 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/BulkMuteFindings.java:60]: should we not catch it and print out any error msg?

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java (hunk 2 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/CreateMuteRule.java:67]: Should we still catch it and then provide some text saying that something went wrong and an exception was thrown?

**Review by stephaniewang526 — APPROVED** [review]

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+    }
+  }
+
+  public static Finding createFinding(String sourceName, String findingId) throws IOException {
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+
+      Instant eventTime = Instant.now();
+
+      // The resource this finding applies to. The Cloud Security Command Center UI can link
+      // the findings for a resource to the corresponding Asset of a resource
```
Comment (catherinerosesantos) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:137]: Re: Asset

Asset --> asset

Alternatively, link to the javadoc for the Asset class?

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+
+  // Deletes a mute configuration given its resource name.
+  // Note: Previously muted findings are not affected when a mute config is deleted.
+  public static void deleteMuteRule(String projectId, String muteConfigId) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (SecurityCenterClient client = SecurityCenterClient.create()) {
+      // Use appropriate MuteConfigName methods depending on the type of parent.
+      // (org -> MuteConfigName.ofOrganizationMuteConfigName()
```
Comment (catherinerosesantos) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/DeleteMuteRule.java:49]: Delete the extra ( before "org"

File: samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+    // parent: Use any one of the following resource paths to list mute configurations:
+    //         - organizations/{organization_id}
+    //         - folders/{folder_id}
+    //         - projects/{project_id}
+    String parentPath = String.format("projects/%s", "your-google-cloud-project-id");
+    listMuteRules(parentPath);
+  }
+
+  // Listing mute configs at the organization level will return all the configs
+  // at the org, folder and project levels.
```
Comment (catherinerosesantos) [inline samples/snippets/src/main/java/com/google/cloud/examples/securitycenter/snippets/muteconfig/ListMuteRules.java:40]: Very small nit: use a serial comma 

org, folder, and project

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+  @BeforeClass
+  public static void setUp() throws IOException {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ID");
+    requireEnvVar("SCC_PROJECT_ORG_ID");
+
+    // Create mute rules.
+    CreateMuteRule.createMuteRule(String.format("projects/%s", PROJECT_ID), MUTE_RULE_CREATE);
+    CreateMuteRule.createMuteRule(String.format("projects/%s", PROJECT_ID), MUTE_RULE_UPDATE);
+    // Create Source.
```
Comment (catherinerosesantos) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:88]: source

File: samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+    SOURCE = createSource(ORGANIZATION_ID);
+    // Create Findings within the source.
```
Comment (catherinerosesantos) [inline samples/snippets/src/test/java/com/google/cloud/examples/securitycenter/snippets/MuteFindingIT.java:90]: findings (just for consistency)

**Review by catherinerosesantos — APPROVED** [review]
Looks great, thanks!

**Pull request merged**